### ISSUE TITLE
Replace &rest with &va-rest in situations where it makes sense

### ISSUE THREAD

**Issue by drmeister** [clasp-developers/clasp#249]
&rest conses a list of the remaining arguments.   core:&va-rest is an optimization that uses the C++ va_list construct to bind something like a list of the remaining arguments to a variable passed as an argument to core:&va-rest.  core:&va-rest is like SB-INT:&MORE. core:&va-rest is completely stack based so it's faster than &rest. core:&va-rest needs to be used carefully, it is not allowed to escape from the function to which the lambda list that contains &va-rest belongs.   A few examples of safe use of core:&va-rest is to pass it as the last argument to APPLY.   

Another use would be to pass them as the last argument to LIST*.

This issue is a recommendation to look through every use of &rest in the C++ code and change it to core:&va-rest in cases where the core:&va-rest value cannot escape.

**Comment by drmeister** [comment]
Related to this issue is the hacks that I had to make to Cleavir to support &va-rest.

beach has a lambda-list class that I had to add a REST-NAME slot to to store the synonym of &rest (&body, &va-rest) that is being used.  Most of the changes were made in sicl/Code/Cleavir/Code-utilities/lambda-lists.lisp

Then everywhere where he checks the keyword against '(&rest &body) I added core:&va-rest
Then when he generates a lambda-list from this lambda-list class I use the REST-NAME slot to recover the name and inject it into the lambda-list.  This last change was made in sicl/Code/Cleavir/Generate-AST/generate-ast.lisp
